File: swift/Neetcode/Arrays/Easy/1189.MaximumBalloons.swift
// Given a string text, you want to use the characters of text 
// to form as many instances of the word "balloon" as possible.
// You can use each character in text at most once. 

func maxNumberOfBalloons(_ text: String) -> Int {
    let textDict = NSCountedSet(array: Array(text))
    let balloon = NSCountedSet(array: Array("balloon"))
    
    var result = Int.max

    for letter in balloon {
        result = min(result, textDict.count(for: letter) / balloon.count(for: letter))
    }

    return result
}

print(maxNumberOfBalloons("nlaebolko"))
print(maxNumberOfBalloons("loonbalxballpoon"))
print(maxNumberOfBalloons("leetcode"))

File: python/Neetcode/Arrays/Easy/1189.MaximumBalloons.py
# Given a string text, you want to use the characters of text 
# to form as many instances of the word "balloon" as possible.
# You can use each character in text at most once. 

from collections import Counter

def maxNumberOfBalloons(text):
    text_dict = Counter(text)
    balloon = Counter('balloon')

    solution = float('inf')

    for letter in balloon:
        solution = min(solution, text_dict[letter]//balloon[letter])

    return solution

print(maxNumberOfBalloons("nlaebolko"))
print(maxNumberOfBalloons("loonbalxballpoon"))
print(maxNumberOfBalloons("leetcode"))